CONTRIBUTING.md

```diff
@@ -0,0 +1,18 @@
+# Contributing to deal.II
+deal.II is a community project that lives by the participation of its members — i.e., including you! It is our goal to build an inclusive and participatory community so we are happy that you are interested in participating.
+
+## Getting started with git and GitHub
+If you are new to using git or the GitHub platform you may find it helpful to review [lecture 32.8](http://www.math.tamu.edu/~bangerth/videos.676.32.8.html), which should be enough to help you get started with using GitHub or possibly contributing to the library itself.
+
+## Asking and answering questions about the library
+The deal.II community maintains an active mailing list hosted by google groups [here](https://groups.google.com/forum/#!forum/dealii). The mailing list is for questions about the library of all levels.
+
+## Bug reports
+It is a great help to the community if you report any bugs in the library that you may find. We keep track of all open issues with the library [here](https://github.com/dealii/dealii/issues). If you can, please try to include a minimal failing example that can help us reproduce the problem.
+
+## Making changes to the library
+To make a change to deal.II you should create a *fork* (through GitHub) of the library and a separate *branch* (sometimes called feature branches). You can propose that your branch be combined with the rest of the library by opening a *pull request*. This will give a chance for others to review your code. While this seems very formal, keeping all of the code review in one place makes it easier to coordinate changes to the library (and there are usually several people making changes to the library at once). This process is described at length in [lecture 32.8](http://www.math.tamu.edu/~bangerth/videos.676.32.8.html). Please do not hesitate to ask questions about the workflow on the mailing list if you are not sure what to do.
+
+deal.II uses `astyle` with a specified configuration to ensure that the entire library is formatted in a consistent way. The use of `astyle` is documented [here](https://dealii.org/developer/index.html). The GNU style guide, which most editors support, is a decent approximation of the coding style used in deal.II.
```

Comment:
can you link to http://dealii.org/developer/doxygen/deal.II/CodingConventions.html instead?